src/executor.cc

```diff
@@ -40,22 +58,12 @@ Result Executor::Execute(Engine* engine,
   if (script->GetPipelines().size() > 1)
     return Result("only single pipeline supported currently");
 
-  auto* pipeline = script->GetPipelines()[0].get();
-
-  // Compile Shaders
-  for (auto& shader_info : pipeline->GetShaders()) {
-    ShaderCompiler sc(script->GetSpvTargetEnv());
-
-    Result r;
-    std::vector<uint32_t> data;
-    std::tie(r, data) = sc.Compile(shader_info.GetShader(), shader_map);
-    if (!r.IsSuccess())
-      return r;
-
-    shader_info.SetData(std::move(data));
-  }
+  Result r = CompileShaders(script, shader_map);
+  if (!r.IsSuccess())
+    return r;
 
-  Result r = engine->CreatePipeline(pipeline);
+  auto* pipeline = script->GetPipelines()[0].get();
+  r = engine->CreatePipeline(pipeline);
```

Comment:
This is where the driver compiler kicks in.   So I don't think this is what @afd  wanted.

This MR only does the "front end" compilation: from shader in the script down to SPIR-V.

Reply:
How about now?